manifests/charts/istiod-remote/templates/serviceaccount.yaml

```diff
@@ -13,7 +13,7 @@ metadata:
   labels:
     app: istiod
     release: {{ .Release.Name }}
-  {{- if .Values.pilot.serviceAccountAnnotations -}}
+  {{- if .Values.pilot.serviceAccountAnnotations }}
```

Comment:
The istiod chart is also in the scope, please update https://github.com/istio/istio/tree/master/manifests/charts/istio-control/istio-discovery and run `make gen`

Reply:
okay

Reply:
@zirain  can you be more brief ?, I'm just a beginner

Reply:
seems there's no such tests, let's do it in an sperated PR

Reply:
@zirain gen test will fail in this case